python/cudf/cudf/core/column/categorical.py

```diff
@@ -1341,23 +1343,13 @@ def find_last_value(self, value: ScalarLike, closest: bool = False) -> int:
         """
         return self.as_numerical.find_last_value(self._encode(value))
 
-    @property
+    @cached_property
```

Comment:
This could be a potential source of bugs. Columns are (unfortunately) not immutable:

```python
In [2]: s = cudf.Series([1, 2, 3])

In [3]: s._column.data.ptr
Out[3]: 140178299551744

In [4]: s[:] = 4

In [5]: s._column.data.ptr
Out[5]: 140178299551744

In [6]: s
Out[6]:
0    4
1    4
2    4
dtype: int64
```

Reply:
I assumed (since the current implement was caching this property anyway) that this property was already being correctly invalidated by such operations, but yes probably safer to remove that entirely.